packages/server/graphql/mutations/createJiraTaskIntegration.ts

```diff
@@ -69,15 +72,55 @@ export default {
       )
     }
 
-    const viewerAuth = await dataLoader.get('freshAtlassianAuth').load({teamId, userId: viewerId})
-    if (!viewerAuth) {
-      return standardError(new Error('The assignee does not have access to Jira'), {
+    const [viewerAuth, assigneeAuth, team] = await Promise.all([
+      dataLoader.get('freshAtlassianAuth').load({teamId, userId: viewerId}),
+      dataLoader.get('freshAtlassianAuth').load({teamId, userId}),
+      dataLoader.get('teams').load(teamId)
+    ])
+    const auth = assigneeAuth ?? viewerAuth
+    if (!auth) {
+      return standardError(new Error('Neither you nor the assignee has access to Jira'), {
         userId: viewerId
       })
     }
+    const teamMembers = await dataLoader.get('teamMembersByTeamId').load(teamId)
+    const viewer = teamMembers.find(({userId}) => userId === viewerId)
+    const assignee = teamMembers.find((user) => user.userId === userId)
+
+    if (!viewer || !assignee) {
```

Comment:
just like the gh file, these are guaranteed to be true when their values are set, no need to re-validate the already validated data